backend/controllers/attendance/attendanceStatistics.js

```diff
@@ -326,9 +327,140 @@ const attendanceStatisticsFactory = (oauthApi, elite2Api, whereaboutsApi, logErr
     }
   }
 
+  const attendanceStatisticsSuspendedList = async (req, res) => {
+    const { agencyId, period, fromDate, toDate } = req.query || {}
+
+    const formattedFromDate = switchDateFormat(fromDate, 'DD/MM/YYYY')
+    const formattedToDate = toDate ? switchDateFormat(toDate, 'DD/MM/YYYY') : switchDateFormat(fromDate, 'DD/MM/YYYY')
+    const currentPeriod = getCurrentPeriod(moment().format())
+    const today = moment().format('DD/MM/YYYY')
+    const periods = period.split('_')
+
+    try {
+      const [user, caseloads, roles] = await Promise.all([
+        oauthApi.currentUser(res.locals),
+        elite2Api.userCaseLoads(res.locals),
+        oauthApi.userRoles(res.locals),
+      ])
+
+      const { activeCaseLoad, activeCaseLoadId } = extractCaseLoadInfo(caseloads)
+
+      if (!period || !fromDate) {
+        return res.redirect(
+          `${attendanceReasonStatsUrl}/suspended?agencyId=${activeCaseLoadId}&period=${currentPeriod}&fromDate=${today}&toDate=${today}`
+        )
+      }
+
+      const scheduledActivities = await elite2Api.getOffenderActivitiesOverDateRange(res.locals, {
+        agencyId,
+        fromDate: formattedFromDate,
+        toDate: formattedToDate,
+        period: period === 'AM_PM' ? '' : period,
+      })
+
+      const suspendedActivites = scheduledActivities.filter(activity => activity.suspended)
+      const totalOffenders = new Set(suspendedActivites.map(activity => activity.bookingId)).size
+
+      // Create mapping between date and a set of bookings with suspended activities on that date
+      // This mapping will be used to get the relevant attendances for each date
+      const suspendedBookingsPerEventDate = {}
```

Comment:
Might be more consistent to use reduce and the spread operator here.

Reply:
I was kind of hoping to leave it a bit less... clever? Whatever the word is, just because it's kind of involved and like this tells the story a bit more clearly. But happy to refactor if you think the opposite :D